curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/use-capture-groups-to-search-and-replace.english.md

```diff
@@ -38,10 +38,12 @@ Write a regex so that it will search for the string <code>"good"</code>. Then up
 tests:
   - text: You should use <code>.replace()</code> to search and replace.
     testString: assert(code.match(/\.replace\(.*\)/));
-  - text: Your regex should change <code>"This sandwich is good."</code> to <code>"This sandwich is okey-dokey."</code>
-    testString: assert(result == "This sandwich is okey-dokey." && replaceText === "okey-dokey");
+  - text: Your regex should change <code>"one two three"</code> to <code>"three two one"</code>
+    testString: assert(result === "three two one");
   - text: You should not change the last line.
     testString: assert(code.match(/result\s*=\s*huhText\.replace\(.*?\)/));
+  - text: You should use at least three capture groups.
+    testString: assert(code.match(/fixRegex\s*=\s*\/(\(.*?\).*?){3,}\//));
```

Comment:
The regex for your match would incorrectly pass the one test if someone used the following code.  There are only two capture groups in the following regex.
```js
let fixRegex = /(\w+)\(\w+\)(\w+)/;
```

Courtesy of [Stack Overflow](https://stackoverflow.com/questions/16046620/regex-to-count-the-number-of-capturing-groups-in-a-regex), we can use the following assert:
```suggestion
    testString: assert((new RegExp(fixRegex.source + '|')).exec('').length - 1 >= 3);
```

Reply:
That's an interesting approach.